src/main/kotlin/org/rust/cargo/toolchain/tools/Cargo.kt

```diff
@@ -162,8 +164,12 @@ class Cargo(toolchain: RsToolchainBase, useWrapper: Boolean = false)
             .stdout
             .dropWhile { it != '{' }
         try {
+            val srcPathConverter: (String) -> String = { path ->
+                if (isBazelPath(path)) { bazelPathToProjectPath(path, projectDirectory.parent.toString()) }
+                else toolchain.toLocalPath(path)
+            }
             return Gson().fromJson(json, CargoMetadata.Project::class.java)
-                .convertPaths(toolchain::toLocalPath)
+                .convertPaths(toolchain::toLocalPath, srcPathConverter)
```

Comment:
e.g. if there is a Cargo manifest at `rust-poc/bazel-bin/lib1/Cargo.toml`, and it contains `[lib] path = "lib1.rs"`, tell IntelliJ that there is a matching source file at `rust-poc/lib1/lib1.rs`